autoreject/autoreject.py

```diff
@@ -507,6 +507,8 @@ def _get_epochs_interpolation(self, epochs, drop_log,
                     peaks[non_picks] = -np.inf
                     # find channels which are bad by rejection threshold
                     interp_chs_mask = drop_log[epoch_idx] == 1
+                    # ignore good channels
+                    peaks[~interp_chs_mask] = -np.inf
```

Comment:
See L515. That is what ignores the good channels

Reply:
Yes, it indeed ignores good channels. But without the change I am proposing, the previous lines pick the first n_interpolate without considering if channels are good or bad. So, if some good channels are within the "worst" n_interpolate, at the end it will result with less bad channels interpolated (because good are ignored, as you noticed).

You can check this by looking at the fix_log at the end: in some cases, it will have less interpolated channels than n_interpolate, even when there are more bad channels.

I'll see if I can add an example from public data.

Reply:
I see what you mean now. Can you confirm it is equivalent to [this](https://github.com/autoreject/autoreject/blob/43811c3c22f5a9b28a5e6f6c03fc58aba9e6b2ca/autoreject/autoreject.py#L446). @dengemann okay with you?

Reply:
Yes, the final result seems equivalent to me.